chore: update README for new client generation to use Cloud Drop and fix workflow scripts

alicejli opened the pull request:
Fixes #9334 

We are moving away from the DevRel services pages to using Cloud Drop for the relevant fields to create new client libraries.

For more context, see b/271301352 and go/cloud-drop.

This PR also fixes a minor bug in the github workflow scripts.

suztomo commented on generation/new_client/README.md:
```diff
@@ -127,52 +127,82 @@ $ gh auth login
 
-Find the short description of the API. Usually the first sentence in the product
-document above.
+The corresponding value in the Cloud Drop page is `documentation.summary` or `documentation.overview`.
+If both of those fields are missing, take the description from the product page above.
+
```
Shall we continue to use the first sentence of the description? This example looks verbose.

alicejli replied:
Makes sense. Updated!

suztomo commented on generation/new_client/README.md:
```diff
@@ -127,52 +127,74 @@ $ gh auth login
+  --proto-path=google/cloud/alloydb \
+  --name-pretty="AlloyDB API" \
+  --product-docs="https://cloud.google.com/alloydb/docs" \
+  --api-description="AlloyDB for PostgreSQL is an open source-compatible database service that
+    provides a powerful option for migrating, modernizing, or building
+    commercial-grade applications. It offers full compatibility with standard
```
Use first sentence.

alicejli replied:
Oops, sorry I missed this one. Thanks for catching - updated.

meltsufin reviewed (changes requested):
Are the other unrelated changes intentional?

alicejli commented:
> Are the other unrelated changes intentional?

Ah yes those checks were originally failing (e.g. https://github.com/googleapis/google-cloud-java/actions/runs/4681749947/jobs/8807412324?pr=9336) so I've updated them here as well. Updated the title/description to reflect those changes as well. Thanks!

suztomo commented on generation/new_client/README.md:
```diff
@@ -124,55 +124,73 @@ $ sudo apt-get install gh
+Example: `alloydb`
 
 ### Proto path
 
-The script takes "proto path" parameter. This is path from google3's root to the
+The script takes "proto path" parameter. This is path from google3's root (`google3/third_party/googleapis/stable/`) to the
 directory that contains versions (e.g., "v1" or "v2"). For example, if we
 have 
 
-> Link to protos: http://google3/google/api/apikeys/v2/apikeys.proto
+> Link to protos: http://google3/third_party/googleapis/stable/google/cloud/alloydb/v1/service.proto.
 
```
Do they have "third_party/googleapis/stable" these days?

alicejli replied:
great point! Updated.

suztomo commented on generation/new_client/README.md:
```diff
@@ -124,55 +124,73 @@ $ sudo apt-get install gh
 You will run new-client.py script with the following parameters.
-Collect them from the ticket before running the command.
+These parameters will be available in the Cloud Drop link (a YAML file) included in the buganizer request.
+The example in this README will AlloyDB's [Cloud Drop](https://github.com/googleapis/googleapis/blob/master/google/cloud/alloydb/v1/alloydb_v1.yaml) file as an example.
```
```suggestion
The example in this README uses AlloyDB's [Cloud Drop](https://github.com/googleapis/googleapis/blob/master/google/cloud/alloydb/v1/alloydb_v1.yaml) file as an example.
```

suztomo commented on generation/new_client/README.md:
```diff
@@ -124,55 +124,73 @@ $ sudo apt-get install gh
 have 
 
-> Link to protos: http://google3/google/api/apikeys/v2/apikeys.proto
+> Link to protos: https://github.com/googleapis/googleapis/blob/master/google/cloud/alloydb/v1/service.proto.
```
This does not match the description in b/271166569. Where did this come?

suztomo replied:
You would need an instruction to convert the "Link to protos: " value in the ticket to the proto path.

suztomo commented on generation/new_client/README.md:
```diff
@@ -143,11 +143,11 @@ Example: `alloydb`
 ### Proto path
 
-The script takes "proto path" parameter. This is path from google3's root (`google3/third_party/googleapis/stable/`) to the
+The script takes "proto path" parameter. This is path from `googleapis/googleapis/blob/master`' root to the
```
Can you clarify what is googleapis/googleapis/blob/master?

alicejli replied:
Yep - it's the Github mirror of `google3/third_party/googleapis/stable`. I updated since we are converting this doc to use all external references.

suztomo commented on generation/new_client/README.md:
```diff
@@ -124,55 +124,75 @@ $ sudo apt-get install gh
+For example, if the buganizer ticket includes:
 
-> Link to protos: http://google3/google/api/apikeys/v2/apikeys.proto
+> Link to protos: `http://depot/google3/third_party/googleapis/stable/google/cloud/alloydb/v1alpha/alloydb_v1alpha.yaml`.
```
```suggestion
> Link to protos: `http://...(omit).../stable/google/cloud/alloydb/v1alpha/alloydb_v1alpha.yaml`.
```

alicejli replied:
updated

suztomo commented on generation/new_client/README.md:
```diff
@@ -124,55 +124,75 @@ $ sudo apt-get install gh
-The script takes "proto path" parameter. This is path from google3's root to the
-directory that contains versions (e.g., "v1" or "v2"). For example, if we
-have 
+The script takes "proto path" parameter. This is path from the internal `google3/third_party/googleapis/stable` root to the
```
This doesn't have to be the internal location. It can be "https://github.com/googleapis/googleapis". We then remove the reference to the google3.

alicejli replied:
I clarified this a bit more; let me know what you think. Thanks!

suztomo reviewed (approved):
Thank you
